sympy/physics/mechanics/kane.py

```diff
@@ -117,7 +136,7 @@ class KanesMethod(_Methods):
     def __init__(self, frame, q_ind, u_ind, kd_eqs=None, q_dependent=None,
             configuration_constraints=None, u_dependent=None,
             velocity_constraints=None, acceleration_constraints=None,
-            u_auxiliary=None, bodies=None, forcelist=None):
+            u_auxiliary=None, bodies=None, forcelist=None, explicit_kinematics=True):
```

Comment:
the way we've implemented it know, explicit_kinematics can be "hot-swapped" as both the implicit and explicit forms are available.
This is because we are only doing the "2nd" part of the original PR of providing the mass/forcing matrices for kinematics in implicit form, i.e. this has no impact on how the dynamics (Kane's) equations themselves are constructed.

I've dropped it from the initializer for now, and we can later add it to the `kanes_equations()` call when the "1" part of the original PR is eventually implemented (i.e. where it can no longer be hotswapped as it would now have an impact on how the dynamics are computed)
I remember the argument for putting it in `kanes_equations()`, which makes sense regarding the "1st" part of the original PR, but since kinematic equations are available before  `kanes_equations()` is invoked, I think it still makes sense to set it in the initial constructor. Perhaps in the future we can decide that `kanes_equations()` can also take it as an argument to overwrite the original value if so desired.


For now I'm tempted to make it a 'private' member so as to signal that it should not be changed mid-run (even though that'd the only way to do it right now if we remove it from the initializer) and to remove it from the initializer and kick the can down the road to figure this out based on how the "1st" part of the original PR is ultimately implemented.

Thoughts?

Reply:
Making it a private member is indeed more future safe. As for where we can set the argument, I'm with either of them. `kanes_equations` is probably the most logical location from a users perspective, but `__init__` is the location where we actually need it. So I would prefere making it a private member and putting it in either the `__init__` or `kanes_equations`. Biggest disadvantage I see here is that it is an extra argument, especially in the `__init__` case.

@moorepants what is your opinion?

Reply:
I don't have any issues with adding new arguments to `__init__`. The KanesMethod class does alot already, of course, but it seems in line to let it also return the explicit form of the kinematic equations.

Reply:
So is the PR fine as is? If so I'll leave to one of you to please merge.
Otherwise please lmk what is left for us to do.

Reply:
@moorepants Based on your comments I assumed that you do agree with the current implementation, i.e. `explicit_kinematics` as attribute and as kwarg in the `__init__`, which is also fine with me.
Since I have no more comments on the PR (all suggestions have been taken into account), can this PR be merged?

Reply:
bump - any chance you can merge please @moorepants ?
Thanks :)